pkg/auth/cache/auth_cache.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	"sync/atomic"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+// stores data
+type AuthCache struct {
+	data atomic.Pointer[map[rbacv1.Subject][]corev1.Namespace]
```

Comment:
I'm thinking we may need a `sync.RWMutex` instead of an atomic pointer swap to synchronize access, but I'm not 100% sure.

Reply:
until access is readonly we should be fine without a mutex, right?

Reply:
Correct, but long-term I think it's what we should probably be using to synchronize accesses.